tools/actions/create-release.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+set -xe
+
+RELEASE_DATE=$1
+RELEASE_LINE=$2
+
+if [ -z "$RELEASE_DATE" ] || [ -z "$RELEASE_LINE" ]; then
+  echo "Usage: $0 <RELEASE_DATE> <RELEASE_LINE>"
+  exit 1
+fi
+
+git config --local user.email "github-bot@iojs.org"
+git config --local user.name "Node.js GitHub Bot"
+
+git node release --prepare --skipBranchDiff --yes --releaseDate "$RELEASE_DATE"
+# We use it to not specify the branch name as it changes based on
+# the commit list (semver-minor/semver-patch)
+git config push.default current
+git push
+
+TITLE=$(awk "/^## ${RELEASE_DATE}/ { print substr(\$0, 4) }" "doc/changelogs/CHANGELOG_V${RELEASE_LINE}.md")
+
+# Use a temporary file for the PR body
+TEMP_BODY=$(mktemp)
+awk "/## ${RELEASE_DATE}/,/^<a id=/{ if (!/^<a id=/) print }" "doc/changelogs/CHANGELOG_V${RELEASE_LINE}.md" > "$TEMP_BODY"
+
+gh pr create --title "$TITLE" --body-file "$TEMP_BODY"
+
+rm "$TEMP_BODY"
```

Comment:
Why use a temp file instead of stdin?
```suggestion
awk "/## ${RELEASE_DATE}/,/^<a id=/{ if (!/^<a id=/) print }" "doc/changelogs/CHANGELOG_V${RELEASE_LINE}.md" |\
gh pr create --title "$TITLE" -F -
```

Reply:
I tried using stdin, gh pr doesn't read from it but attempted to read using an fs syscall. I was using this command but `gh help` was being printed instead.

`--body-file` option expects a filename not a raw piped input.

Reply:
It seems a bug, I was running exactly this command:

```
➜  node (v23.3.0-proposal) awk '/## 2024-11-18/,/^<a id=/{ if (!/^<a id=/) print }' doc/changelogs/CHANGELOG_V23.md | gh pr create --body-file -
```

It's printing gh help instead

Reply:
Also, slightly unrelated: should we open it as draft?

Reply:
I think there's no need to open it as a draft as we want the CI to run as fast as possible

Reply:
The thing is awk expects a single-quoted strings for regex parsing, I can test, but I suspect your suggestion won't work as well.

Reply:
I could reproduce using `sh`

```
$ RES=$(awk "/## ${RELEASE_DATE}/,/^<a id=/{ if (!/^<a id=/) print }" "doc/changelogs/CHANGELOG_V${RELEASE_LINE}.md")
$ echo $RES
## 2024-11-11, Version 23.2.0 (Current), @aduh95 ### Notable Changes #### Update root certificates to NSS 3.104 This is the version of NSS that shipped in Firefox 131.0 on 2024-10-01. Certificates added: BSDmakefile BUILD.gn BUILDING.md CHANGELOG.md CODE
```

but it seems line breaks aren't passing as expected. I'm not sure, I'm fine either way, I just feel the temporary file is way easier to read/maintain.

Reply:
You need to use quotes:

```console
$ RELEASE_DATE=2024-07-17 
$ RELEASE_LINE=22
$ RES="$(awk "/## ${RELEASE_DATE}/,/^<a id=/{ if (!/^<a id=/) print }" "doc/changelogs/CHANGELOG_V${RELEASE_LINE}.md")"
$ echo "$RES"
## 2024-07-17, Version 22.5.0 (Current), @RafaelGSS prepared by @aduh95

### Notable Changes

* \[[`1367c5558e`](…
```

Reply:
> I think there's no need to open it as a draft as we want the CI to run as fast as possible

Yeah but the commit message needs amending anyway, plus the notable changes will need some work.